flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -400,9 +404,23 @@ protected void processInput(MailboxDefaultAction.Controller controller) throws E
             controller.allActionsCompleted();
             return;
         }
-        CompletableFuture<?> jointFuture = getInputOutputJointFuture(status);
-        MailboxDefaultAction.Suspension suspendedDefaultAction = controller.suspendDefaultAction();
-        assertNoException(jointFuture.thenRun(suspendedDefaultAction::resume));
+
+        final Suspension suspension = controller.suspendDefaultAction();
```

Comment:
I have shortened it while you were doing the review, when I was changing the semantic of `idleTimeMsPerSecond` metric. Can you change the fixup commit if it's good enough?

Reply:
I think it can be shortened further by inlining gauges and removing if:
```
assertNoException(inputProcessor.getAvailableFuture()
    .thenRun(new ResumeWrapper(
        suspension,
        recordWriter.isAvailable()
                ? ioMetrics.getIdleTimeMsPerSecond()
                : ioMetrics.getBackPressuredTimePerSecond())));
```
(the gauge is still computed at the time of creation of callback)
WDYT?

Reply:
I have simplified it, but in your example you missed that sometimes we need to wait on `inputProcessor.getAvailableFuture()` and other times on `recordWriter.getAvailableFuture()`

Reply:
Thanks for fixing it. I think the current version still under-estimates idle time if both input and output are consistently unavailable. But probably that's fine.